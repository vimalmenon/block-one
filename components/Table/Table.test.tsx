import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import { Table, TableFooter, TableRow } from './'

describe('Table Component', () => {
  test('Table: Snapshot testing with Footer', () => {
    const columns = [
      {
        header: 'Column 1',
        accessor: 'one',
        templete: ({ index }: any) => index + 1,
      },
      {
        header: 'Column 2',
        accessor: 'two',
      },
    ]
    const items = [
      {
        one: 1,
        two: 'First Row',
      },
      {
        one: 2,
        two: 'Second Row',
      },
    ]
    const footer = [
      {
        value: 'This is footer 1',
        colSpan: 2,
      },
      {
        value: 'This is footer 2',
      },
    ]
    const wrapper = shallow(<Table columns={columns} items={items} footer={footer} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  test('Table: Snapshot testing without Footer', () => {
    const columns = [
      {
        header: 'Column 1',
        accessor: 'one',
        templete: ({ index }: any) => index + 1,
      },
      {
        header: 'Column 2',
        accessor: 'two',
      },
    ]
    const items = [
      {
        one: 1,
        two: 'First Row',
      },
      {
        one: 2,
        two: 'Second Row',
      },
    ]
    const wrapper = shallow(<Table columns={columns} items={items} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  test('Table: Table Row click event Testing', () => {
    const columns = [
      {
        header: 'Column 1',
        accessor: 'one',
        templete: ({ index }: any) => index + 1,
      },
      {
        header: 'Column 2',
        accessor: 'two',
      },
    ]
    const items = [
      {
        one: 1,
        two: 'First Row',
      },
      {
        one: 2,
        two: 'Second Row',
      },
    ]
    const onRowClick = jest.fn()
    const wrapper = shallow(<Table columns={columns} items={items} onRowClick={onRowClick} />)
    wrapper.find('tr').at(1).simulate('click')
    expect(onRowClick).toHaveBeenCalled()
  })
  test('TableFooter: Snapshot Testing', () => {
    const footer = [
      {
        value: 'This is footer 1',
        colSpan: 2,
      },
      {
        value: 'This is footer 2',
      },
    ]
    const wrapper = shallow(<TableFooter footers={footer} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  test('TableRow: Snapshot Testing with template', () => {
    const column = {
      header: 'Column 1',
      accessor: 'one',
      templete: ({ index }: any) => index + 1,
    }
    const item = {
      one: 1,
      two: 'First Row',
    }
    const wrapper = shallow(<TableRow column={column} item={item} index={1} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  test('TableRow: Snapshot Testing without template', () => {
    const column = {
      header: 'Column 1',
      accessor: 'one',
    }
    const item = {
      one: 1,
      two: 'First Row',
    }
    const wrapper = shallow(<TableRow column={column} item={item} index={1} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
