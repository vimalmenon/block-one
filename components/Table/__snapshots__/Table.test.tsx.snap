// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Component Table: Snapshot testing with Footer 1`] = `
<table
  className="w-full rounded border border-blue-400"
>
  <thead>
    <tr
      className="bg-blue-500 text-white"
    >
      <th
        className="text-left p-2 font-medium"
        key="0"
      >
        Column 1
      </th>
      <th
        className="text-left p-2 font-medium"
        key="1"
      >
        Column 2
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      className="text-gray-600 bg-gray-50"
      key="0"
      onClick={[Function]}
    >
      <TableRow
        column={
          Object {
            "accessor": "one",
            "header": "Column 1",
            "templete": [Function],
          }
        }
        index={0}
        item={
          Object {
            "one": 1,
            "two": "First Row",
          }
        }
        key="0"
      />
      <TableRow
        column={
          Object {
            "accessor": "two",
            "header": "Column 2",
          }
        }
        index={0}
        item={
          Object {
            "one": 1,
            "two": "First Row",
          }
        }
        key="1"
      />
    </tr>
    <tr
      className="text-gray-600 bg-blue-50"
      key="1"
      onClick={[Function]}
    >
      <TableRow
        column={
          Object {
            "accessor": "one",
            "header": "Column 1",
            "templete": [Function],
          }
        }
        index={1}
        item={
          Object {
            "one": 2,
            "two": "Second Row",
          }
        }
        key="0"
      />
      <TableRow
        column={
          Object {
            "accessor": "two",
            "header": "Column 2",
          }
        }
        index={1}
        item={
          Object {
            "one": 2,
            "two": "Second Row",
          }
        }
        key="1"
      />
    </tr>
  </tbody>
  <TableFooter
    footers={
      Array [
        Object {
          "colSpan": 2,
          "value": "This is footer 1",
        },
        Object {
          "value": "This is footer 2",
        },
      ]
    }
  />
</table>
`;

exports[`Table Component Table: Snapshot testing without Footer 1`] = `
<table
  className="w-full rounded border border-blue-400"
>
  <thead>
    <tr
      className="bg-blue-500 text-white"
    >
      <th
        className="text-left p-2 font-medium"
        key="0"
      >
        Column 1
      </th>
      <th
        className="text-left p-2 font-medium"
        key="1"
      >
        Column 2
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      className="text-gray-600 bg-gray-50"
      key="0"
      onClick={[Function]}
    >
      <TableRow
        column={
          Object {
            "accessor": "one",
            "header": "Column 1",
            "templete": [Function],
          }
        }
        index={0}
        item={
          Object {
            "one": 1,
            "two": "First Row",
          }
        }
        key="0"
      />
      <TableRow
        column={
          Object {
            "accessor": "two",
            "header": "Column 2",
          }
        }
        index={0}
        item={
          Object {
            "one": 1,
            "two": "First Row",
          }
        }
        key="1"
      />
    </tr>
    <tr
      className="text-gray-600 bg-blue-50"
      key="1"
      onClick={[Function]}
    >
      <TableRow
        column={
          Object {
            "accessor": "one",
            "header": "Column 1",
            "templete": [Function],
          }
        }
        index={1}
        item={
          Object {
            "one": 2,
            "two": "Second Row",
          }
        }
        key="0"
      />
      <TableRow
        column={
          Object {
            "accessor": "two",
            "header": "Column 2",
          }
        }
        index={1}
        item={
          Object {
            "one": 2,
            "two": "Second Row",
          }
        }
        key="1"
      />
    </tr>
  </tbody>
</table>
`;

exports[`Table Component TableFooter: Snapshot Testing 1`] = `
<tfoot>
  <tr
    className="bg-gray-500 text-white"
  >
    <td
      className="p-2"
      colSpan={2}
      key="0"
    >
      This is footer 1
    </td>
    <td
      className="p-2"
      colSpan={1}
      key="1"
    >
      This is footer 2
    </td>
  </tr>
</tfoot>
`;

exports[`Table Component TableRow: Snapshot Testing with template 1`] = `
<td
  className="p-2"
>
  1
</td>
`;

exports[`Table Component TableRow: Snapshot Testing without template 1`] = `
<td
  className="p-2"
>
  1
</td>
`;
